akka-http/src/main/scala/scala/util/matching/JavaConverters.scala

```diff
@@ -0,0 +1,10 @@
+package scala.util.matching
```

Comment:
@ktoso See the comment. I added this in order to turn a Java `Pattern` into a Scala `Regex` without recompiling it. That's pretty important, since many directives will be invoked from within lambdas that are called on every route, and you don't want to do the conversion + recompilation inside the directive for every call. 

But scala's `Regex` constructor is package-private, so that's why I put it in this package. I know it's a hack, and I'm open for alternatives that give the same performance without sacrificing DSL comfort level. Maybe adding a directive to the Scala DSL that takes a plain Java `Pattern`?